packages/@react-spectrum/table/test/TreeGridTable.test.tsx

```diff
@@ -0,0 +1,1518 @@
+          triggerPress(chevronCell);
+          checkSelection(onSelectionChange, [
+            'Row 1 Lvl 1'
+          ]);
+          checkRowSelection(rows.slice(0, 1), true);
+
+          onSelectionChange.mockReset();
+          triggerPress(chevron);
+          expect(onSelectionChange).not.toHaveBeenCalled();
+          checkRowSelection(rows.slice(0, 1), true);
+        });
+
+        it('should select a nested row on click', function () {
+          let treegrid = render(<ManyRowsExpandableTable onSelectionChange={onSelectionChange} selectionMode="multiple" selectionStyle="checkbox" disabledKeys={null} onAction={null} />);
+          let rowgroups = treegrid.getAllByRole('rowgroup');
+          let rows = within(rowgroups[1]).getAllByRole('row');
+          let cell = getCell(treegrid, 'Row 1, Lvl 3, Foo');
+
+          checkRowSelection(rows, false);
+          triggerPress(cell);
+          checkSelection(onSelectionChange, [
+            'Row 1 Lvl 3'
+          ]);
+          checkRowSelection(rows.slice(2, 3), true);
+        });
+      });
+
+      describe('with keyboard', function () {
+        it('should select a nested row by pressing the Enter key on a row', function () {
+          let treegrid = render(<ManyRowsExpandableTable onSelectionChange={onSelectionChange} selectionMode="multiple" selectionStyle="checkbox" disabledKeys={null} onAction={null} />);
+          let rowgroups = treegrid.getAllByRole('rowgroup');
+          let rows = within(rowgroups[1]).getAllByRole('row');
+
+          checkRowSelection(rows, false);
+          pressWithKeyboard(rows[1], 'Enter');
+          checkSelection(onSelectionChange, [
+            'Row 1 Lvl 2'
+          ]);
+          checkRowSelection(rows.slice(1, 2), true);
+          checkSelectAll(treegrid);
+        });
+
+        it('should select a nested row by pressing the Space key on a row', function () {
+          let treegrid = render(<ManyRowsExpandableTable onSelectionChange={onSelectionChange} selectionMode="multiple" selectionStyle="checkbox" disabledKeys={null} onAction={null} />);
+          let rowgroups = treegrid.getAllByRole('rowgroup');
+          let rows = within(rowgroups[1]).getAllByRole('row');
+
+          checkRowSelection(rows, false);
+          pressWithKeyboard(rows[1]);
+          checkSelection(onSelectionChange, [
+            'Row 1 Lvl 2'
+          ]);
+          checkRowSelection(rows.slice(1, 2), true);
+          checkSelectAll(treegrid);
+        });
+
+        it('should select a row by pressing the Enter key on a chevron cell', function () {
+          let treegrid = render(<ManyRowsExpandableTable onSelectionChange={onSelectionChange} selectionMode="multiple" selectionStyle="checkbox" disabledKeys={null} onAction={null} />);
+          let rowgroups = treegrid.getAllByRole('rowgroup');
+          let rows = within(rowgroups[1]).getAllByRole('row');
+          let cell = getCell(treegrid, 'Row 1, Lvl 1, Foo');
+
+
```

Comment:
I've noticed a couple double spaces like this in the tests. The tests that are very comprehensive. :)